gcp-bundle/NOTICE

```diff
@@ -29,72 +28,106 @@ from the source code management (SCM) system project uses.
 
 --------------------------------------------------------------------------------
 
-NOTICE for Group: commons-codec  Name: commons-codec  Version: 1.15
+Apache Commons Codec
+Copyright 2002-2024 The Apache Software Foundation
 
-src/test/org/apache/commons/codec/language/DoubleMetaphoneTest.java
-contains test data from http://aspell.net/test/orig/batch0.tab.
-Copyright (C) 2002 Kevin Atkinson (kevina@gnu.org)
+--------------------------------------------------------------------------------
 
-===============================================================================
+gRPC
+Copyright 2014 The gRPC Authors
 
-The content of package org.apache.commons.codec.language.bm has been translated
-from the original php source code available at http://stevemorse.org/phoneticinfo.htm
-with permission from the original authors.
-Original source copyright:
-Copyright (c) 2008 Alexander Beider & Stephen P. Morse.
+This product contains a modified portion of 'OkHttp', an open source
+HTTP & SPDY client for Android and Java applications, which can be obtained
+at:
 
---------------------------------------------------------------------------------
+  * LICENSE:
+    * okhttp/third_party/okhttp/LICENSE (Apache License 2.0)
+  * HOMEPAGE:
+    * https://github.com/square/okhttp
+  * LOCATION_IN_GRPC:
+    * okhttp/third_party/okhttp
 
-NOTICE for Group: io.grpc  Name: grpc-netty-shaded  Version: 1.55.1
+This product contains a modified portion of 'Envoy', an open source
+cloud-native high-performance edge/middle/service proxy, which can be
+obtained at:
 
-                          The Netty Project
```

Comment:
Why was the Netty notice content removed? Netty is still included in the dependencies according to LICENSE.

Reply:
It's coming via grpc (shaded). 

Imho, it should be in the NOTICE: when you bundle a dependency that shade another one, it should be mention in our NOTICE.

Reply:
Ok I went back and forth on this but from my reading of https://infra.apache.org/licensing-howto.html#bundled-vs-non-bundled it does seem like we should explicitly include Netty so I agree with the current change. Seems like whether it's shaded or not, as long as those bits exist in the gcp-bundle, which they do, they must be in the NOTICE.